File: Examples/ClassDiagram/Ruby/podcast.rb
require_relative 'media_item'
require_relative 'playable'

class Podcast < MediaItem
  include Playable

  attr_accessor :host, :episode_number

  def initialize(title, duration, host, episode_number)
    super(title, duration)
    @host = host
    @episode_number = episode_number
  end
end

File: Examples/ClassDiagram/CSharp/Podcast.cs
using System;

namespace MediaLibrary;

public class Podcast : MediaItem {
    private readonly string _host;
    private readonly int _episodeNumber;

    public string Host => _host;
    public int EpisodeNumber => _episodeNumber;

    public Podcast(string title, double duration, string host, int episodeNumber) 
        : base(title, duration, Genre.Talk) {
        _host = host;
        _episodeNumber = episodeNumber;
    }

    public override void Play() {
        Console.WriteLine($"Playing podcast episode {_episodeNumber}: {Title} hosted by {_host}");
    }
}
